network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/NetworkImpl.java

```diff
@@ -858,6 +856,10 @@ void invalidateComponents() {
                             .build());
             updateResource();
         }
+        if (type == CgmesControlAreas.class) {
+            resource.getAttributes().setCgmesControlAreas(new CgmesControlAreasAttributes());
```

Comment:
You should initialize 'controlAreas' field in the CgmesControlAreasAttributes object created, using the 'extension' parameter provided to the method (like the other extensions above)

Reply:
There is nothing to init here, unlike the other extension, the extension created through the adder is empty. Then we can get a control area adder from the control areas extension to add a new control area, but there is no way in extension adder to directly add a control area
```java
network.newExtension(CgmesControlAreasAdder.class)
// no method available
                    .add();
```

Reply:
ok